selfhost/codegen.jakt

```diff
@@ -4489,6 +4489,17 @@ struct CodeGenerator {
         let expression_type = .program.get_type(expr.type())
         let name = call.name_for_codegen()
 
+        if name.as_name_for_use() == "operator[]" {
+            guard call.args.size() == 1 else {
```

Comment:
how about
```suggestion
        guard name.as_name_for_use() != "operator[]" or call.args.size() != 1 else {
```

to allow `operator[]` with more than 1 argument? support for this is not landing in C++ any time soon, but why panic when we can just not.

Reply:
I guess I could just write all the arguments separated by commas into the braces, then.

Reply:
nvm, I will just use the equivalence where it's currently supported and leave the rest to be an `operator[](x, y)` call as you suggested.

Reply:
Using `guard` to ensure the `return` maybe needs some naming to be easily parsed.

Reply:
> I guess I could just write all the arguments separated by commas into the braces, then.

Ah that's currently incorrect (`x[foo, bar, baz]` is just `x[baz]` with whatever side effects `foo` and `bar` have).
calling `operator[](foo, bar, baz)` is the "correct" approach here.

Reply:
And sure, the `if` is fine too, equivalent anyway.